templates/flavors/aks/cluster-template.yaml

```diff
@@ -29,7 +29,7 @@ metadata:
   name: ${CLUSTER_NAME}-control-plane
 spec:
   subscriptionID: ${AZURE_SUBSCRIPTION_ID}
-  resourceGroup: "${AZURE_RESOURCE_GROUP}"
+  resourceGroup: "${AZURE_RESOURCE_GROUP:=$CLUSTER_NAME}"
   location: "${AZURE_LOCATION}"
```

Comment:
What do folks think about a location default? (applies to all the rest of the flavors too)

Reply:
that sounds good to me

Reply:
the reason I see for not doing that is that location does require you to have quota in that region so making the user set it is a good way for them to choose a location where they can actually deploy VMs... Otherwise it's very easy to shoot yourself in the foot as a new user and get a quota failure on your first deployment (which right now isn't handled very gracefully -- see https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/519)

Reply:
I'm good with the vars that are currently defaulted. I just wanted to push a bit to how aggressive / conservative we think we should be.

Reply:
I'd find location defaulting somewhat annoying, personally -- I'd rather get validation failure than accidentally create a cluster in the wrong region